nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/PasswordHelper.java

```diff
@@ -139,7 +139,8 @@ public static boolean verifyPassword(String password, String hashedPassword) {
     public static byte[] digestWithSalt(String password, byte[] salt, String algorithm) {
         try {
             MessageDigest md = MessageDigest.getInstance(algorithm);
-            md.update(password.getBytes("UTF-8"));
+            String temp_password = (password == null ? "" : password);
```

Comment:
Please rename your variable to `temp_password` (we use camel case).
Remove parenthesis between the ternary expression.